pyramid/scripts/pserve.py

```diff
@@ -48,92 +47,106 @@ class PServeCommand(object):
     """
     default_verbosity = 1
 
-    parser = optparse.OptionParser(
-        usage,
+    parser = argparse.ArgumentParser(
+        usage="%(prog)s config_uri [var=value]",
         description=textwrap.dedent(description)
         )
-    parser.add_option(
+    parser.add_argument(
         '-n', '--app-name',
         dest='app_name',
         metavar='NAME',
         help="Load the named application (default main)")
-    parser.add_option(
+    parser.add_argument(
         '-s', '--server',
         dest='server',
         metavar='SERVER_TYPE',
         help="Use the named server.")
-    parser.add_option(
+    parser.add_argument(
         '--server-name',
         dest='server_name',
         metavar='SECTION_NAME',
         help=("Use the named server as defined in the configuration file "
               "(default: main)"))
-    parser.add_option(
+    parser.add_argument(
         '--reload',
         dest='reload',
         action='store_true',
         help="Use auto-restart file monitor")
-    parser.add_option(
+    parser.add_argument(
         '--reload-interval',
         dest='reload_interval',
         default=1,
         help=("Seconds between checking files (low number can cause "
               "significant CPU usage)"))
-    parser.add_option(
+    parser.add_argument(
         '-b', '--browser',
         dest='browser',
         action='store_true',
         help="Open a web browser to server url")
-    parser.add_option(
+    parser.add_argument(
         '-v', '--verbose',
         default=default_verbosity,
         dest='verbose',
         action='count',
         help="Set verbose level (default " + str(default_verbosity) + ")")
-    parser.add_option(
+    parser.add_argument(
         '-q', '--quiet',
         action='store_const',
         const=0,
         dest='verbose',
         help="Suppress verbose output")
+    parser.add_argument(
+        'config_uri',
+        nargs='?',
+        default=None,
+        help='The URI to the configuration file.',
+        )
+    parser.add_argument(
+        'config_args',
+        nargs='*',
+        default=(),
+        help='Arbitrary options to override those in the [app:main] section '
```

Comment:
Nope... not `app:main` section. Should say something more like `Variables required by the config file. For example, http_port=%%(http_port)s would expect "http_port=8080" to be passed here.`.